engines/wage/combat.cpp

```diff
@@ -0,0 +1,914 @@
+
+			if (scene != NULL && scene->_chrs.empty()) {
+				directions |= (1 << dir);
+			}
+		}
+	}
+
+	return directions;
+}
+
+void WageEngine::regen() {
+	Chr *player = _world->_player;
+	int curHp = player->_context._statVariables[PHYS_HIT_CUR];
+	int maxHp = player->_context._statVariables[PHYS_HIT_BAS];
+	int delta = maxHp - curHp;
+
+	if (delta > 0) {
+		int bonus = (int)(delta / (8 + _rnd->getRandomNumber(2)));
+		player->_context._statVariables[PHYS_HIT_CUR] += bonus;
+	}
+}
+
+void WageEngine::takeObj(Obj *obj) {
+	if ((int)_world->_player->_inventory.size() >= _world->_player->_maximumCarriedObjects) {
+		appendText("Your pack is full, you must drop something.");
+	} else {
+		char buf[256];
+
+		_world->move(obj, _world->_player);
+		int type = _world->_player->wearObjIfPossible(obj);
+		if (type == Chr::HEAD_ARMOR) {
+			snprintf(buf, 256, "You are now wearing the %s.", obj->_name.c_str());
+			appendText(buf);
+		} else if (type == Chr::BODY_ARMOR) {
+			snprintf(buf, 256, "You are now wearing the %s.", obj->_name.c_str());
+			appendText(buf);
+		} else if (type == Chr::SHIELD_ARMOR) {
+			snprintf(buf, 256, "You are now wearing the %s.", obj->_name.c_str());
+			appendText(buf);
+		} else if (type == Chr::MAGIC_ARMOR) {
+			snprintf(buf, 256, "You are now wearing the %s.", obj->_name.c_str());
+			appendText(buf);
+		} else {
+			snprintf(buf, 256, "You now have the %s.", obj->_name.c_str());
+			appendText(buf);
+		}
+		appendText(obj->_clickMessage.c_str());
+	}
+}
+
+bool WageEngine::handleMoveCommand(Directions dir, const char *dirName) {
+	Scene *playerScene = _world->_player->_currentScene;
+	const char *msg = playerScene->_messages[dir].c_str();
+
+	if (!playerScene->_blocked[dir]) {
+		int destX = playerScene->_worldX + directionsX[dir];
+		int destY = playerScene->_worldY + directionsY[dir];
+
+		Scene *scene = _world->getSceneAt(destX, destY);
+
+		if (scene != NULL) {
+			if (strlen(msg) > 0) {
+				appendText(msg);
+			}
+			_world->move(_world->_player, scene);
+			return true;
+		}
+	}
+	if (strlen(msg) > 0) {
+		appendText(msg);
+	} else {
+		Common::String txt("You can't go ");
+		txt += dirName;
+		txt += ".";
+		appendText(txt.c_str());
+	}
+
+	return true;
+}
+
+bool WageEngine::handleLookCommand() {
+	appendText(_world->_player->_currentScene->_text.c_str());
+
+	Common::String *items = getGroundItemsList(_world->_player->_currentScene);
+	if (items != NULL) {
+		appendText(items->c_str());
+
+		delete items;
+	}
+
+	return true;
+}
+
+Common::String *WageEngine::getGroundItemsList(Scene *scene) {
+	ObjArray objs;
+
+	for (ObjList::const_iterator it = scene->_objs.begin(); it != scene->_objs.end(); ++it)
+		if ((*it)->_type != Obj::IMMOBILE_OBJECT)
+			objs.push_back(*it);
+
+	if (objs.size()) {
+		Common::String *res = new Common::String("On the ground you see ");
+		appendObjNames(*res, objs);
+		return res;
+	}
+	return NULL;
+}
+
+void WageEngine::appendObjNames(Common::String &str, ObjArray &objs) {
```

Comment:
This doesn't seem to modify contents of `objs`, thus I suggest making it `const`. I noticed `getIndefiniteArticle`, takes a non-`const` reference to `Common::String`, but since it doesn't change its contents either, it can easily be modified to take a `const` reference too, allowing this method (and maybe others) to use `const` in more places too.